clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -526,128 +611,193 @@ scheduler_tmpl_t<RecordType, ReaderType>::advance_region_of_interest(input_info_
+    VPRINT(this, 3, "post-skip: adding %" PRId64 " instrs to %" PRId64 "\n",
+           input.reader->get_instruction_ordinal() - input_start_instr,
+           stream.cur_instr_count_);
+    stream.cur_instr_count_ +=
+        input.reader->get_instruction_ordinal() - input_start_instr;
+    // If we skipped from the start we may not have seen the initial headers:
+    // use the input's cached copies.
+    if (stream.version_ == 0) {
+        stream.version_ = input.reader->get_version();
+        stream.last_timestamp_ = input.reader->get_last_timestamp();
+        stream.filetype_ = input.reader->get_filetype();
+        stream.cache_line_size_ = input.reader->get_cache_line_size();
+        stream.chunk_instr_count_ = input.reader->get_chunk_instr_count();
+        stream.page_size_ = input.reader->get_page_size();
+    }
+    // We let the user know we've skipped.  There's no discontinuity for the
+    // first one so we do not insert a marker there (if we do want to insert one,
+    // we need to update the view tool to handle a window marker as the very
+    // first entry).
+    if (input.cur_region > 0) {
+        input.queue.push(create_region_separator_marker(input.tid, input.cur_region));
+    }
+    return sched_type_t::STATUS_SKIPPED;
+}
+
+template <typename RecordType, typename ReaderType>
+typename scheduler_tmpl_t<RecordType, ReaderType>::stream_status_t
+scheduler_tmpl_t<RecordType, ReaderType>::pick_next_input(int output_ordinal)
+{
+    if (options_.mapping == MAP_TO_RECORDED_OUTPUT) {
+        // TODO i#5843: For now we have a single input; we'll need to change this
+        // once we move the file_reader_t interleaving here.
+        return sched_type_t::STATUS_EOF;
+    }
+    if (options_.mapping != MAP_TO_CONSISTENT_OUTPUT) {
+        // TODO i#5843: Implement synthetic scheduling with instr/time quanta.
+        // These will require locks and central coordination.
+        return sched_type_t::STATUS_NOT_IMPLEMENTED;
+    }
+    int index = outputs_[output_ordinal].cur_input;
+    while (true) {
+        if (index < 0) {
+            // We're done with the prior thread; take the next one that was
+            // pre-allocated to this output (pre-allocated to avoid locks).  Invariant:
+            // the same output_ordinal will not be accessed by two different threads
+            // simultaneously in this mode, allowing us to support a lock-free
+            // parallel-friendly increment here.
+            ++outputs_[output_ordinal].input_indices_index;
+            if (outputs_[output_ordinal].input_indices_index >=
+                static_cast<int>(outputs_[output_ordinal].input_indices.size())) {
+                VPRINT(this, 2, "next_record[%d]: all at eof\n", output_ordinal);
+                return sched_type_t::STATUS_EOF;
+            }
+            index = outputs_[output_ordinal]
+                        .input_indices[outputs_[output_ordinal].input_indices_index];
```

Comment:
Wow, I think this section of code might be easier to read if you put outputs_[output_ordinal].input_indices_index into a (well-named) temp variable when you increment it, and then use that for the rest of the code here.

Reply:
Done, using `indices_index`.